x-pack/plugins/stack_connectors/server/connector_types/gemini/gemini.test.ts

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { GeminiConnector } from './gemini';
+import { ServiceParams } from '@kbn/actions-plugin/server';
+import { Config, Secrets, RunActionParams } from '../../../common/gemini/types';
+import { ActionsConfigurationUtilities } from '@kbn/actions-plugin/server/actions_config';
+import { Logger } from '@kbn/logging';
+import type { Services } from '@kbn/actions-plugin/server/types';
+import {
+  KibanaRequest,
+  SavedObjectsClientContract,
+  ElasticsearchClient,
+  KibanaRequestEvents,
+  IKibanaSocket,
+} from '@kbn/core/server';
+import { ConnectorTokenClient } from '@kbn/actions-plugin/server/lib/connector_token_client';
+import type { Headers } from 'packages/core/http/core-http-server/src/router/headers';
+import { AxiosResponse, AxiosHeaders } from 'axios';
+import { RunApiResponse } from '../../../common/gemini/types';
+import { RunApiResponseSchema } from '../../../common/gemini/schema';
+
+jest.mock('@kbn/actions-plugin/server/sub_action_framework/helpers/validators', () => ({
+  assertURL: jest.fn(),
+}));
+
+jest.mock('@kbn/actions-plugin/server', () => {
+  const originalModule = jest.requireActual('@kbn/actions-plugin/server');
+
+  return {
+    ...originalModule,
+    SubActionConnector: jest.fn().mockImplementation(
```

Comment:
curious why you mock `SubActionConnector`. Is this a pattern you saw somewhere else?

Reply:
If you are able, please follow the testing pattern in `x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.test.ts`

Reply:
hey @stephmilovic , @shivankawasthi11 would be the best person to answer this. as he worked on the unit tests. @shivankawasthi11 could you please look into this?

Reply:
@stephmilovic Will add an implementation instead of mocking it